src/main/java/com/es/phoneshop/browsing/SortingComparator.java

```diff
@@ -1,11 +1,8 @@
-package com.es.phoneshop.comparator;
+package com.es.phoneshop.browsing;
```

Comment:
Regarding packages:
1) Rename browsing package to comparator.
2) move enums from constant to new wnum package
3) service classes should be in service package and dao classes in dao package. They are functionaly different.
4) Move listeners in listener package
5) If you have internal packages in model for each type of entity then create package for Product and HistoryRecord. I would put all entities in the root of model package. It's up to you